File: educational-core/src/com/jetbrains/edu/coursecreator/actions/placeholder/CCCreateAnswerPlaceholderDialog.java
package com.jetbrains.edu.coursecreator.actions.placeholder;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.DialogWrapper;
import com.intellij.openapi.ui.ValidationInfo;
import com.intellij.openapi.util.text.StringUtil;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class CCCreateAnswerPlaceholderDialog extends DialogWrapper {

  private static final String TITLE_SUFFIX = " Answer Placeholder";
  private final CCAddAnswerPlaceholderPanel myPanel;

  public CCCreateAnswerPlaceholderDialog(@NotNull final Project project, @NotNull String placeholderText, boolean isEdit) {
    super(project, true);

    myPanel = new CCAddAnswerPlaceholderPanel(placeholderText);
    String title = (isEdit ? "Edit" : "Add") + TITLE_SUFFIX;
    setTitle(title);
    String buttonText = (isEdit ? "OK" : "Add");
    setOKButtonText(buttonText);
    init();
    initValidation();
  }

  @NotNull
  public String getTaskText() {
    return StringUtil.notNullize(myPanel.getAnswerPlaceholderText()).trim();
  }

  @Nullable
  @Override
  protected JComponent createCenterPanel() {
    return myPanel;
  }

  @Nullable
  @Override
  public ValidationInfo doValidate() {
    return null;
  }

  @Nullable
  @Override
  public JComponent getPreferredFocusedComponent() {
    return myPanel.getPreferredFocusedComponent();
  }
}

File: educational-core/src/com/jetbrains/edu/coursecreator/actions/placeholder/CCCreateAnswerPlaceholderDialog.kt
package com.jetbrains.edu.coursecreator.actions.placeholder

import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.DialogWrapper
import com.intellij.openapi.ui.ValidationInfo
import com.jetbrains.edu.coursecreator.CCUtils.DEFAULT_PLACEHOLDER_TEXT
import com.jetbrains.edu.learning.courseFormat.AnswerPlaceholder
import com.jetbrains.edu.learning.messages.EduCoreBundle
import javax.swing.JComponent

open class CCCreateAnswerPlaceholderDialog(
  project: Project,
  placeholder: AnswerPlaceholder
) : DialogWrapper(project, true) {

  private val panel: CCAddAnswerPlaceholderPanel = CCAddAnswerPlaceholderPanel(placeholder.placeholderText ?: DEFAULT_PLACEHOLDER_TEXT)

  init {
    title = EduCoreBundle.message("ui.dialog.create.answer.placeholder.add")
    val buttonText = EduCoreBundle.message("label.add")
    setOKButtonText(buttonText)
    super.init()
    initValidation()
  }

  override fun createCenterPanel(): JComponent {
    return panel
  }

  override fun doValidate(): ValidationInfo? {
    return null
  }

  override fun getPreferredFocusedComponent(): JComponent? {
    return panel.getPreferredFocusedComponent()
  }

  open fun getPlaceholderText(): String = panel.getAnswerPlaceholderText().trim()

}
